CONTRIBUTING.md

```diff
@@ -254,3 +254,50 @@ reviewer.
 For public PRs which do not have a preassigned reviewer, a TFX engineer will
 monitor them and perform initial triage within 5 business days. But such
 contributions should be trivial (i.e, documentation fixes).
+
+## Continuous Integration
+
+This project makes use of CI for
+
+- Building the `tfx` python package when releases are made
+- Running tests
+- Linting pull requests
+- Building documentation
+
+These four _workflows_ trigger automatically when certain _events_ happen.
+
+### Pull Requests
+
+When a PR is made:
+
+- Wheels and an sdist are built using the code in the PR branch. Multiple wheels
+  are built for a [variety of architectures and python
+  versions](https://github.com/tensorflow/tfx/blob/master/.github/workflows/wheels.yml).
+  If the PR causes any of the wheels to fail to build, the failure will be
+  reported in the checks for the PR.
+
+- Tests are run via `pytest`. If a test fails, the workflow failure will be
+  reported in the checks for the PR.
+
+- Lint checks are run on the changed files. This workflow makes use of the
+  `.pre-commit-config.yaml`, and if any lint violations are found the workflow
+  reports a failure on the list of checks for the PR.
+
+If the author of the PR makes a new commit to the PR branch, these checks are
+run again on the new commit.
+
+### Releases
+
+When a release is made on GitHub the workflow that builds wheels runs, just as
```

Comment:
Does the release process also cover the image upload in Docker or GCP AR?

Reply:
No, the other stuff that isn't related to building python wheels isn't included. The docker image is as far as I can tell just a tool for making an environment in which the `tfx` wheels can be built. Is that right?

If so, I'd caution against becoming overreliant on docker images for building wheels. I've seen cases (notably other TF-adjacent packages) which can _only_ be built inside certain docker images, which makes library development very cumbersome; `pip install -e .` becomes impossible if the only way to build the project is inside a container.